packages/medusa/src/services/order-edit.ts

```diff
@@ -769,6 +801,35 @@ export default class OrderEditService extends TransactionBaseService {
       orderEdit.confirmed_at = new Date()
       orderEdit.confirmed_by = context.loggedInUserId
 
+      if (orderEdit.payment_collection) {
+        if (
+          orderEdit.payment_collection.status !==
+          PaymentCollectionStatus.AUTHORIZED
+        ) {
+          throw new MedusaError(
+            MedusaError.Types.NOT_ALLOWED,
+            "Unable to complete an order edit if the payment is not authorized"
+          )
+        }
+
+        for (const payment of orderEdit.payment_collection.payments) {
+          await this.paymentProviderService_.updatePayment(payment.id, {
+            order_id: orderEdit.order_id,
+          })
+        }
+      } else {
+        const total = await this.getTotals(orderEdit.id)
+        if (total.difference_due < 0) {
+          await this.orderService_
+            .withTransaction(manager)
+            .createRefund(
+              orderEdit.order_id,
+              total.difference_due * -1,
+              "Order Edit Difference"
+            )
+        }
+      }
+
```

Comment:
todo: move this to API layer - should be possible to handle OrderEdits completely separate from payment collections

Reply:
That is somewhat related to the previous comment. If we don't manage that relation in here, at which point the payment collection will relate to the original order?
And in case the difference is negative, we force the refund from the original order payment to be manually called?

Reply:
Similar idea as mentioned above and even more important here actually, since the confirmation can either come from the customer or be forced by an admin.

If the admin forces the confirmation, it will be their responsibility to align the payment later. If the customer confirms, then we want to ensure that they have, in fact, already paid. 

So would move this check into `packages/medusa/src/api/routes/store/order-edits/complete-order-edit.ts`.

Think there was an commented snippet in there already:
```ts

    // TODO once payment collection is done
    const paymentCollection = await this.paymentCollectionService_.withTransaction(manager).retrieve(orderEdit.payment_collection_id)
    if (!paymentCollection.authorized_at) {
      throw new MedusaError(
        MedusaError.Types.NOT_ALLOWED,
        "Unable to complete an order edit if the payment is not authorized"
      )
    }

```

The refund will either be a manual action performed by the admin user or if people want to automate this they could build a subscriber, but it shouldn't block the confirmation possibility if this fails:

```js
eventBusService.subscribe("order-edit.confirmed", ({ id }) => {
  // 1. get order
  // 2. create a refund
})
```